dbt_semantic_interfaces/transformations/default_granularity.py

```diff
@@ -34,7 +38,7 @@ def transform_model(semantic_manifest: PydanticSemanticManifest) -> PydanticSema
                 continue
 
             default_granularity = TimeGranularity.DAY
-            seen_agg_time_dimensions: Set[TimeDimensionReference] = set()
+            seen_agg_time_dimensions: Set[Tuple[SemanticModel, TimeDimensionReference]] = set()
```

Comment:
Im always wary of using larger objects in sets, so if we could change this to like a semanticmodelreference would be nice

Reply:
Good call